### PR TITLE
keep error traces

### PR DESCRIPTION
**Pull request by zeitlinger**

### CODE REVIEW

File: custom/src/main/java/com/grafana/extensions/sampler/DynamicSampler.java [hunk 1]
```diff
@@ -74,6 +77,15 @@ public Set<String> getSampledTraces() {
   }
 
   boolean shouldSample(ReadableSpan span) {
+    return isSlow(span) || hasError(span);
+  }
+
+  private boolean hasError(ReadableSpan span) {
+    return span.toSpanData().getStatus().getStatusCode() == StatusCode.ERROR
+        || span.getAttributes().get(EXCEPTION) != null;
```
Comment (s4kh) [inline custom/src/main/java/com/grafana/extensions/sampler/DynamicSampler.java:85]: Inside attributes, I don't see any key with `exception.type`. Should we use `http.response.status_code` - change key to `error.type`?
```
{client.address="127.0.0.1", error.type="500", http.request.method="GET", http.response.status_code=500, http.route="/checkout", network.peer.address="127.0.0.1", network.peer.port=45596, network.protocol.version="1.1", server.address="localhost", server.port=8082, thread.id=58, thread.name="http-nio-8082-exec-9", url.path="/checkout", url.scheme="http", user_agent.original="Java/21.0.5"}
```

Comment (zeitlinger) [reply custom/src/main/java/com/grafana/extensions/sampler/DynamicSampler.java:85]: "exception.type" is automatically added when a java exception is thrown

usually, the status code is also set to ERROR - which should also be the case for the span you added above

Comment (s4kh) [reply custom/src/main/java/com/grafana/extensions/sampler/DynamicSampler.java:85]: Right after the error is thrown, I see `error.type` but not `exception.type` in the span attributes. That is why I changed key to `error.type` in [here](https://github.com/grafana/grafana-opentelemetry-java/pull/834/commits/9e3edafa1a37a1d9d127543afd11e46ebe8a0d87), should I discard the change?

Comment (zeitlinger) [reply custom/src/main/java/com/grafana/extensions/sampler/DynamicSampler.java:85]: https://opentelemetry.io/docs/specs/semconv/exceptions/exceptions-spans/#exception-event

that's the one I mean - but you're also right: https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-client

so we should probably check for both

Comment (s4kh) [reply custom/src/main/java/com/grafana/extensions/sampler/DynamicSampler.java]: added it.

**Review by s4kh — APPROVED** [review]

**Pull request merged**